node/router/src/heartbeat.rs

```diff
@@ -107,6 +107,12 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
             return;
         }
 
+        let is_validator = self.router().node_type().is_validator();
+        // Skip if the node is not requesting peers.
+        if is_validator && !self.router().allow_external_peers() {
```

Comment:
nit: is_validator is no longer necessary as part of the check (because allow_external_peers takes precedence)